src/uct/rdmacm/rdmacm_md.c

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "rdmacm_md.h"
+
+#define UCT_RDMACM_MD_PREFIX              "rdmacm"
+#define UCT_RDMACM_ADDR_RESOLVE_TIMEOUT    500    /* ms */
+
+static ucs_config_field_t uct_rdmacm_md_config_table[] = {
+  {"", "", NULL,
+   ucs_offsetof(uct_rdmacm_md_config_t, super), UCS_CONFIG_TYPE_TABLE(uct_md_config_table)},
+
+  {NULL}
+};
+
+static void uct_rdmacm_md_close(uct_md_h md)
+{
+    uct_rdmacm_md_t *rdmacm_md = ucs_derived_of(md, uct_rdmacm_md_t);
+    ucs_free(rdmacm_md);
+}
+
+ucs_status_t uct_rdmacm_md_query(uct_md_h md, uct_md_attr_t *md_attr)
+{
+    md_attr->cap.flags         = UCT_MD_FLAG_SOCKADDR;
+    md_attr->cap.max_alloc     = 0;
+    md_attr->cap.max_reg       = 0;
+    md_attr->rkey_packed_size  = 0;
+    md_attr->reg_cost.overhead = 0;
+    md_attr->reg_cost.growth   = 0;
+    memset(&md_attr->local_cpus, 0xff, sizeof(md_attr->local_cpus));
+    return UCS_OK;
+}
+
+int uct_rdmacm_is_sockaddr_accessible(uct_md_h md, const ucs_sock_addr_t *sockaddr,
+                                      uct_sockaddr_accessibility_t mode)
+{
+    struct rdma_event_channel *event_ch = NULL;
+    struct rdma_cm_id *cm_id = NULL;
+    int is_accessible;
+
+    if ((mode != UCT_SOCKADDR_ACC_LOCAL) && (mode != UCT_SOCKADDR_ACC_REMOTE)) {
+        ucs_fatal("Unknown sockaddr accessibility mode %d", mode);
+        return 0;
+    }
+
+    event_ch = rdma_create_event_channel();
+    if (event_ch == NULL) {
+        ucs_error("Failed to create an rdmacm event channel: %m");
```

Comment:
better phrase the errors as "rdma_create_event_channel() failed: %m" and so on
so would have easy reference to the relevant function
same for bind_addr/resolve_addr below